src/Resolver/AppOutcomesResolver.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal\Resolver;
+
+use OneSignal\Apps;
+use OneSignal\Devices;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class AppOutcomesResolver implements ResolverInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve(array $data): array
+    {
+        return (new OptionsResolver())
+            ->setDefined('outcome_names')
+            ->setAllowedTypes('outcome_names', 'string[]')
+            ->setDefined('outcome_time_range')
+            ->setAllowedTypes('outcome_time_range', 'string')
+            ->setAllowedValues('outcome_time_range', [Apps::OUTCOME_TIME_RANGE_HOUR, Apps::OUTCOME_TIME_RANGE_DAY, Apps::OUTCOME_TIME_RANGE_MONTH])
+            ->setDefault('outcome_time_range', Apps::OUTCOME_TIME_RANGE_HOUR)
+            ->setDefined('outcome_platforms')
+            ->setAllowedTypes('outcome_platforms', 'int[]')
+            ->setAllowedValues('outcome_platforms', static function (array $platforms): bool {
```

Comment:
Why not just directly setting the array of all platforms? :)

Reply:
Because outcome_platforms is an array with integers not a single value. I thought that options resolver cannot do intersection of two arrays

Reply:
Good point, I though it should work but apparently not :)